app/api/modules.py

```diff
@@ -1,26 +1,39 @@
 from flask_rest_jsonapi import ResourceDetail
+from flask import request
+from flask_jwt import current_identity as current_user, _jwt_required
+from flask import current_app as app
 
 from app.api.bootstrap import api
-from app.api.schema.modules import ModuleSchema
+from app.api.schema.modules import ModuleSchemaAdmin
 from app.models import db
 from app.models.module import Module
+from app.api.helpers.exceptions import ForbiddenException
 
 
 class ModuleDetail(ResourceDetail):
     """
-    module detail by id
+    setting detail by id
     """
+
     def before_get(self, args, kwargs):
-        """
-        before get method to get the resource id for fetching details
-        :param args:
-        :param kwargs:
-        :return:
-        """
         kwargs['id'] = 1
 
-    decorators = (api.has_permission('is_admin', methods="PATCH", id="1"),)
+        if 'Authorization' in request.headers:
+            _jwt_required(app.config['JWT_DEFAULT_REALM'])
+
+            if current_user.is_admin or current_user.is_super_admin:
+                self.schema = ModuleSchemaAdmin
+            else:
+                raise ForbiddenException(
+                    {'pointer': '/data/attributes'}, "Access Forbidden"
+                )
+        else:
+            raise ForbiddenException(
+                {'pointer': '/data/attributes'}, "Authorization Required"
+            )
+
+    decorators = (api.has_permission('is_admin', methods='PATCH', id='1'),)
```

Comment:
Why aren't we checking for Admin access for 'GET' request ?

Reply:
@srv-twry 
> Why aren't we checking for Admin access for 'GET' request ?

so that other users can see this but not edit this.
> Please change your commit message before it gets merged. It's a feature that you're implementing right ? If yes, then it should be something like feat: Implement ....

Its is not a feature but a bug in existing API `/v1/modules` 
Check https://github.com/fossasia/open-event-server/issues/4761 I have reported it as bug report in Issue